clang/include/clang/Driver/Options.td

```diff
@@ -8545,6 +8545,11 @@ def dxc_entrypoint : Option<["--", "/", "-"], "E", KIND_JOINED_OR_SEPARATE>,
                      Group<dxc_Group>,
                      Visibility<[DXCOption]>,
                      HelpText<"Entry point name">;
+def dxc_hlsl_version : Option<["/", "-"], "HV", KIND_JOINED_OR_SEPARATE>,
+                     Group<dxc_Group>,
+                     Visibility<[DXCOption]>,
+                     HelpText<"HLSL Version">,
+                     NormalizedValues<["2016", "2017", "2018", "2021", "202x"]>;
```

Comment:
In the current state of this patch I don't think this does anything. I don't think the `Values` or `NormalizedValues` stuff really do anything (other than maybe populating the help spew) if you're not using the marshalling infrastructure (https://clang.llvm.org/docs/InternalsManual.html#option-marshalling-infrastructure).

We don't currently have a `DriverOptions` construct, so driver-only options can't use the marshalling stuff, which is unfortunate.

I think for the purposes of this change we should probably go with the direction in this PR and revisit marshalling later.

Reply:
Populating the help spew seems helpful! I can remove the marshalling part, but it seems helpful to leave the rest as is.